Api Key Rework - PRO-467

ewoolsey opened the pull request:
Shortened api keys to 32 bytes total, 16 for the uuid and 16 for the secret. Also fixed the missing/incorrect attributes for the predefined settings in the config.toml.

Dzejkop commented on src/api_key.rs:
```diff
@@ -3,40 +3,37 @@ use std::str::FromStr;
 pub struct ApiKey {
     pub relayer_id: String,
-    pub api_key: [u8; 32],
+    pub secret: [u8; SECRET_LEN],
```
We actually need to make it into a `Vec<u8>` or some sort of small vec. We already have 32 byte secret keys running in the dev env

ewoolsey replied:
Sorry could you elaborate? You mean you'd like to keep backwards compatibility with 32byte secrets? (48byte total for the api key?)

Dzejkop replied:
Yes, exactly. I see that you already implemented that

ewoolsey commented:
@Dzejkop I've updated this to have the api keys be of variable length. I've also made the internals of `ApiKey` private so that we can ensure the validation of it's length upon construction. Then I've also included some tests to make sure that's being done correctly. Let me know if this looks okay to you.

ewoolsey commented:
I've updated this PR again as I discovered that API keys were being inadvertently revealed in logs in plain text. To remedy this I created a new type, `ApiSecret` and implemented some safety features around it. `ApiKey` no longer implements `Display` and we must explicitly call `ApiKey::reveal` in order to generate the key. `ApiKey` still implements `Debug` but the implementation replaces the secret with `***`
